Options/WeakAurasOptions.lua

```diff
@@ -6960,30 +6960,45 @@ function WeakAuras.CreateFrame()
         elseif(mode == "table") then
           displayStr = WeakAuras.DisplayToTableString(id);
         end
+        importexportbox.editBox:SetMaxBytes(nil);
         importexportbox.editBox:SetScript("OnEscapePressed", function() importexport:Close(); end);
         importexportbox.editBox:SetScript("OnChar", function() importexportbox:SetText(displayStr); importexportbox.editBox:HighlightText(); end);
         importexportbox.editBox:SetScript("OnMouseUp", function() importexportbox.editBox:HighlightText(); end);
-        importexportbox.editBox:SetScript("OnTextChanged", nil);
         importexportbox:SetLabel(id.." - "..#displayStr);
         importexportbox.button:Hide();
         importexportbox:SetText(displayStr);
         importexportbox.editBox:HighlightText();
         importexportbox:SetFocus();
       end
     elseif(mode == "import") then
+      local textBuffer, i, lastPaste = {}, 0, 0
+      local function clearBuffer(self)
+        self:SetScript('OnUpdate', nil)
+          local pasted = strtrim(table.concat(textBuffer))
+          importexportbox.editBox:ClearFocus();
+          pasted = pasted:match( "^%s*(.-)%s*$" );
+          if (#pasted > 20) then
+            WeakAuras.ImportString(pasted);
+            importexportbox:SetLabel(L["Processed %i chars"]:format(i));
+            importexportbox.editBox:SetMaxBytes(10);
+            importexportbox.editBox:SetText("Done");
```

Comment:
So Semlar did something different in his code, the truncates the string at 2500 chars and shows those 2500 via a fontstring overlayed. I found that more confussing then helpful, so I did something different.